src/compiler.rs

```diff
@@ -104,13 +104,8 @@ impl Compiler {
         Ok(codegen(&project, &project.scopes[file_scope_id]))
     }
 
-    pub fn compile(&mut self, fname: &Path) -> Result<(), JaktError> {
-        let cpp_string = self.convert_to_cpp(fname)?;
-
-        let mut out_file = std::fs::File::create("output.cpp")?;
-        out_file.write_all(cpp_string.as_bytes())?;
-
-        Ok(())
+    pub fn compile(&mut self, fname: &Path) -> Result<String, JaktError> {
+        self.convert_to_cpp(fname)
```

Comment:
Now that this function only calls one other function, I'm not sure if it makes sense to actually keep it?

Reply:
The rationale I told myself was that we'd later add other ways to compile, or other output formats, so it made sense to keep the wrapper. But if that's too forward thinking, for sure can remove :)